pages/_document.tsx

```diff
@@ -1,20 +1,32 @@
 import Document, { Html, Head, Main, NextScript } from 'next/document';
+import Script from 'next/script';
 
 import { ReactElement } from 'react';
 
 export default class AppDocument extends Document {
   render(): ReactElement {
     return (
       <Html id="root" lang="en">
-        <Head />
-        <body className="govuk-template__body lbh-template__body js-enabled">
-          <Main />
-          <NextScript />
-          <noscript
+        <Head>
+          <Script
+            id="gtm"
             dangerouslySetInnerHTML={{
-              __html: `<iframe src="https://www.googletagmanager.com/ns.html?id=${process.env.NEXT_PUBLIC_GTM_ID}" height="0" width="0" style="display: none; visibility: hidden;" />`,
+              __html: `(function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({"gtm.start":new Date().getTime(),event:"gtm.js"});var f=d.getElementsByTagName(s)[0],j=d.createElement(s),dl=l!="dataLayer"?"&l="+l:"";j.async=true;j.src="https://www.googletagmanager.com/gtm.js?id="+i+dl;f.parentNode.insertBefore(j,f);})(window,document,"script","dataLayer","${process.env.NEXT_PUBLIC_GTM_ID}");`,
```

Comment:
I don't get it.
What does this line14 do that the block lines17-29 doesn't? :thinking:

Reply:
Clearly these have differences, but this `__html` and the string's that's fed into it roles are unclear.

Reply:
So my understanding is the iframe is a fallback if JS is disabled.

Reply:
That makes sense.
I was thinking that maybe this was duplicated by accident :sweat_smile: